File: backend/src/main/java/com/yunki/lessonpt/location/domain/Location.java
package com.yunki.lessonpt.location.domain;

import java.time.LocalDateTime;

import com.yunki.lessonpt.common.model.RecordStatus;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 강사에게 속한 연습 장소다.
 * 응답으로는 쓰지 않고, 삭제 시각은 서비스 안에서만 다룬다.
 */
@Getter
@Setter
@NoArgsConstructor
public class Location {

    private Long locationId;
    private Long teacherId;
    private String name;
    private Integer displayOrder;
    private String address;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
    private RecordStatus status;
    private LocalDateTime deletedAt;
}

File: frontend/src/types/location.ts
export type Location = {
  locationId: number
  name: string
  displayOrder: number
  address: string | null
  createdAt: string
  updatedAt: string
}

export type LocationWriteBody = {
  name: string
  address: string | null
}

export type StudentLocationAssignment = {
  teacherStudentLocationId: number
  locationId: number
  locationName: string
  address: string | null
}
